fvm/src/call_manager/backtrace.rs

```diff
@@ -96,23 +110,61 @@ pub struct Cause {
     pub message: String,
 }
 
+#[derive(Clone, Debug)]
+pub struct FatalCause {
+    /// The alternate-formatted message from the anyhow error.
+    pub error_msg: String,
+    /// The backtrace, captured if the relevant
+    /// [environment variables](https://doc.rust-lang.org/std/backtrace/index.html#environment-variables) are enabled.
+    pub backtrace: String,
+}
+
+/// The ultimate "cause" of a failed message.
+#[derive(Clone, Debug)]
+pub enum Cause {
+    /// The original cause was a syscall error.
+    Syscall(SyscallCause),
+    /// The original cause was a fatal error.
+    Fatal(FatalCause),
+}
+
```

Comment:
We can collapse these into a single enum.

```rust
/// The ultimate "cause" of a failed message.
#[derive(Clone, Debug)]
pub enum Cause {
    /// The original cause was a syscall error.
    Syscall {
        module: &'static str,
        /// The syscall function name.
        function: &'static str,
        /// The exact syscall error.
        error: ErrorNumber,
        /// The informational syscall message.
        message: String,
    },
    /// The original cause was a fatal error.
    Fatal {
        /// The alternate-formatted message from the anyhow error.
        error_msg: String,
        /// The backtrace, captured if the relevant
        /// [environment variables](https://doc.rust-lang.org/std/backtrace/index.html#environment-variables) are enabled.
        backtrace: String,
    },
}
```

Reply:
I had inline structs in a previous version, but promoted them to types so I could annotate them with `derive` macros. I think I don't need those now, so I can demote them again.